content/collections.md

```diff
@@ -0,0 +1,370 @@
+2. Second you run the migration.
+  - At this point you should be confident that all lists have a `todoCount`.
+3. Finally, you deploy the new code that relies on the new schema and no longer copes with the old schema.
+  - Now we are safe to rely on `list.todoCount` in our UI.
+
+Another thing to be aware of, especially with such multi-stage deploys, is that being prepared to rollback is important! For this reason, the migrations package allows you to specify a `down()` function and set `MIGRATION=x` to migration _back_ to version `x`. 
+
+If you find you need to roll your code version back, you'll need to be careful about the data, and step careful through your deployment steps in reverse.
+
+## Relations between collections
+
+As we discussed earlier, it's very common in Meteor applications to relate documents in different collections. Consequently, it's also very common to need to write queries fetching related documents once you have a document you are interested (for instance all the todos that are on a single list).
+
+To do so, we can attach functions to the prototype of the documents that belong to a given collection, to give us "methods" on the documents (in the object oriented sense). We can then use these methods to create new queries to find related documents.
+
+### Collection Helpers
+
+To do, we can use the [`dburles:collection-helpers`](https://atmospherejs.com/dburles/collection-helpers) to easily attach such methods (or "helpers") to documents. For instance:
+
+```js
+Lists.helpers({
+  // A list is considered to be private if it has a userId set
+  isPrivate() {
+    return !!this.userId;
+  }
+});
+```
+
+Once we've attached this helper to the `Lists` collection, every time we fetch a list from the database (on the client or server), it will have a `.isPrivate()` helper available:
+
+```js
+const list = Lists.findOne();
+if (list.isPrivate()) {
+  console.log('The first list is private!');
+}
+```
+
+### Relational helpers
+
+Now we can attach helpers to documents, it's simple to define a helper that fetches related documents
+
+// XXX: we don't actually have this in the Todos app, but we probably should
+// XXX: is this .todos() or .getTodos() ?
+
+```js
+Lists.helpers({
+  todos() {
+    return Todos.find({listId: this._id}, {sort: {createdAt: -1}});
+  }
+});
+```
+
+Now we can find all the todos for a list easily:
+```js
+const list = Lists.findOne();
+console.log(`The first list has ${list.todos().count()} todos`);
+```
+
+### Query modification
+
+It's not uncommon to need to find a slightly different set of related documents; for instance you might want to find all the incomplete todos on the list. One way to do that would be to add a second helper `list.incompleteTodos` which re-writes the query in `list.todos()`, but it's better to re-use code if possible.
+
+// XXX: this package doesn't exist and the API isn't decided on
```

Comment:
Should we just not have it in the initial guide release? I feel like we've been procrastinating on this hardcore.